smk/gene_abundance.smk

```diff
@@ -654,25 +654,27 @@ rule merge_gene_abund:
 # We use --prefix to prepend before relabeling.
 rule relabel_merged_gene_abund:
     input:
-        metadata=metadata,
-        original="{somewhere}/{something}.raw"
+        original="{wd}/{omics}/9-mapping-profiles/{post_analysis_out}/{label}.p{identity}.{suffix}.raw"
     output:
-        relabeled="{somewhere}/{something}"
+        relabeled="{wd}/{omics}/9-mapping-profiles/{post_analysis_out}/{label}.p{identity}.{suffix}"
+    wildcard_constraints:
+        label="all|genes_abundances",
+        identity='\d+',
+        suffix="|".join(['profile.abund.prop.txt', 'profile.abund.prop.genome.txt', 'profile.relabund.prop.genome.txt', 'bed'])
     shadow:
         "minimal"
     log:
-        "{somewhere}/relabel_{something}.log"
+        "{wd}/logs/{omics}/9-mapping-profiles/{post_analysis_out}/relabel_{label}.p{identity}.{suffix}.log"
     params:
-        prefix = lambda wildcards: "" if (wildcards.something.endswith('.bed')) else "--prefix {}.".format(omics)
+        prefix = lambda wildcards: "" if (wildcards.suffix.endswith('bed')) else "--prefix {}.".format(omics)
     threads: 4
     conda:
         config["minto_dir"]+"/envs/r_pkgs.yml"
     resources:
         mem=30
     shell:
         """
-        time (Rscript {script_dir}/relabel_profiles_using_metadata.R --from sample --to sample_alias --threads {threads} --metadata {input.metadata} --input {input.original} --output relabeled.txt {params.prefix}) >& {log}
-        rsync -a relabeled.txt {output.relabeled}
+        time Rscript {script_dir}/relabel_profiles_using_metadata.R --from sample --to sample_alias --threads {threads} --metadata {metadata} --input {input.original} --output {output.relabeled} {params.prefix} >& {log}
```

Comment:
Writing to local file in shadow directory was intentional. For large text files, it will be better to write on local disk and then rsync.

Reply:
The snakemake docs (https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#shadow-rules) to me suggest that the output in the rule is made in the shadow folder and only moved to the proper place after the rule is done. So the R script should be outputting to the local disk.

Reply:
I remember reading this before and being confused what happens in `shadow: minimal`. I might have tested and found that it directly creates the `{output}` file in the output directory. Now I can't recall the details. Could you please test it? If it indeed writes locally and then moves to `{output}`, there are so many rules where I do it manually that could be updated.

Reply:
`shadow: minimal` with relative paths as input/output works as described in the docs: it writes initially to the shadow folder, then once the cluster job is done it moves the output. However, as we use absolute paths for input/output, the shadow folder is only used if (tmp) files relative to the current path (!) are made.
Thus I'm adding the rsync back.